Library/Homebrew/formula.rb

```diff
@@ -1867,8 +1868,8 @@ def exec_cmd(cmd, args, out, logfn)
       ENV.refurbish_args if setup_py_in_args || setuptools_shim_in_args
     end
 
-    $stdout.reopen(out)
-    $stderr.reopen(out)
+    $stdout = out
+    $stderr = out
     out.close
```

Comment:
Feels weird to be closing this since we're now still using that instance, but it does seem to work? Any `IO` experts able to explain this to me?

Reply:
According to my minimal example here, it really shouldn't work. If I uncomment `w.close`, the fork fails with an `IOError`.

```ruby
r, w = IO.pipe

pid = fork do
  r.close
  $stdout = w
  $stderr = w
  begin
    w.close
    $stdout.puts "STDOUT"
    $stderr.puts "STDERR"
  rescue => e
    STDOUT.puts e.class
    exit! 1
  end
  exit! 0
end

w.close

Process.wait(pid)

puts r.read

r.close
```

Reply:
Ok, it seems like `$stdout` and `$stderr` aren't even used, because the `exec` call below will just use `STDOUT` and `STDERR`, apparently. So instead of overriding these variables, use `exec(cmd, *args, err: out, out: out)` below.